service/history/chasm_notifier.go

```diff
@@ -32,7 +33,10 @@ func NewChasmNotifier() *ChasmNotifier {
 // been reached and resubscribe if necessary, while holding a lock on the execution. The caller must
 // arrange for the unsubscribe function to be called when they have finished monitoring the channel
 // for notifications. It is safe to call the unsubscribe function multiple times and concurrently.
-func (n *ChasmNotifier) Subscribe(key chasm.ExecutionKey) (<-chan struct{}, func()) {
+func (n *ChasmNotifier) Subscribe(key chasm.ExecutionKey) (<-chan struct{}, func(), error) {
+	if key.RunID == "" {
+		return nil, nil, serviceerror.NewInvalidArgument("run id is required")
+	}
```

Comment:
We could use a custom non-serviceerror error here if people prefer. Or we could do a pass on making error strategy consistent later.